tests/core/data_layer/util.py

```diff
@@ -189,8 +189,11 @@ def create_valid_node_values(
     right_hash: Optional[bytes32] = None,
 ) -> Dict[str, Any]:
     if node_type == NodeType.INTERNAL:
+        treehash = Program.to((left_hash, right_hash)).get_tree_hash_precalc(
+            *[h for h in [left_hash, right_hash] if h is not None]
+        )
```

Comment:
`get_tree_hash_precalc()` takes varargs of `bytes32`, not `Optional[bytes32]`. This creates a `List[Optional[bytes32]]` the condenses it into a `List[bytes32]` and then expands it to a parameter pack.

Reply:
This probably needs an overload.  If the node type is internal then the hashes must be specified, otherwise they must not be.  I believe the sql layer already enforces this, fwiw, but yeah, hinting was incomplete at this level.

Reply:
you mean `get_tree_hash_precalc()` needs an overload?
Would you prefer I change this function to take a vararg of `Optional[bytes32]`?

Reply:
No, this function we're in `create_valid_node_values()`.  If called for an internal node the hashes should be specified.  If called for a terminal node, they should not be specified.  It might be that it should just be two functions.  The suggested overload would help call sites where it is called with a literal for the node type, not sure about call sites using a variable there.  Probably not those cases.  And, I don't think it would actually help anything at all inside the function where you are trying to fix something.  So...

Maybe just assert not `None` on the hashes here.  Maybe also assert `None` for them in the terminal node case below for completeness.  Not beautiful, but puts checks in according to the intended usage and also the hinted usage if the overload is done.

Reply:
in the `TERMINAL` node case, `left_hash` (and I assume `right_hash`) are not guaranteed to be `None` at least not in the tests. In the internal node case, the tests don't seem to trigger an assert however.

e.g.
```
[gw0] linux -- Python 3.11.6 /home/arvid/dev/2/chia-blockchain/venv/bin/python
venv/lib/python3.11/site-packages/anyio/pytest_plugin.py:72: in wrapper
    yield runner.run_fixture(func, kwargs)
venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py:1926: in run_fixture
    retval = self.get_loop().run_until_complete(
/usr/lib/python3.11/asyncio/base_events.py:653: in run_until_complete
    return future.result()
venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py:1896: in _call_in_runner_task
    return await future
venv/lib/python3.11/site-packages/anyio/_backends/_asyncio.py:1874: in _run_tests_and_fixtures
    retval = await coro
tests/core/data_layer/conftest.py:92: in valid_node_values_fixture
    return create_valid_node_values(node_type=node_type, left_hash=node_a.hash, right_hash=node_b.hash)
tests/core/data_layer/util.py:203: in create_valid_node_values
    assert left_hash is None
E   AssertionError
```